mlflow/tracking/fluent.py

```diff
@@ -207,27 +210,29 @@ def start_run(
-                     and metrics under that run. The run's end time is unset and its status
-                     is set to running, but the run's other attributes (``source_version``,
-                     ``source_type``, etc.) are not changed.
+        and metrics under that run. The run's end time is unset and its status
+        is set to running, but the run's other attributes (``source_version``,
+        ``source_type``, etc.) are not changed.
     :param experiment_id: ID of the experiment under which to create the current run (applicable
-                          only when ``run_id`` is not specified). If ``experiment_id`` argument
-                          is unspecified, will look for valid experiment in the following order:
-                          activated using ``set_experiment``, ``MLFLOW_EXPERIMENT_NAME``
-                          environment variable, ``MLFLOW_EXPERIMENT_ID`` environment variable,
-                          or the default experiment as defined by the tracking server.
-    :param run_name: Name of new run.
-                     Used only when ``run_id`` is unspecified. If a new run is created and
-                     ``run_name`` is not specified, a unique name will be generated for the run.
+        only when ``run_id`` is not specified). If ``experiment_id`` argument
+        is unspecified, will look for valid experiment in the following order:
+        activated using ``set_experiment``, ``MLFLOW_EXPERIMENT_NAME``
+        environment variable, ``MLFLOW_EXPERIMENT_ID`` environment variable,
+        or the default experiment as defined by the tracking server.
+    :param run_name: Name of new run. Used only when ``run_id`` is unspecified. If a new run is
+        created and ``run_name`` is not specified, a unique name will be generated for the run.
     :param nested: Controls whether run is nested in parent run. ``True`` creates a nested run.
     :param tags: An optional dictionary of string keys and values to set as tags on the run.
-                 If a run is being resumed, these tags are set on the resumed run. If a new run is
-                 being created, these tags are set on the new run.
+        If a run is being resumed, these tags are set on the resumed run. If a new run is
+        being created, these tags are set on the new run.
     :param description: An optional string that populates the description box of the run.
-                        If a run is being resumed, the description is set on the resumed run.
-                        If a new run is being created, the description is set on the new run.
-    :return: :py:class:`mlflow.ActiveRun` object that acts as a context manager wrapping
-             the run's state.
+        If a run is being resumed, the description is set on the resumed run.
+        If a new run is being created, the description is set on the new run.
+    :param log_system_metrics: bool, defaults to False. If True, system metrics will be logged
```

Comment:
Following from https://github.com/mlflow/mlflow/pull/9557/files#r1326679560, I think the default value of this should be `None`, since the user could call `mlflow.enable_system_metrics_logging()` and then call `mlflow.start_run()` without specifying the `log_system_metrics()` kwarg. I would expect the behavior to be "enable system metrics logging for the run", but, if the default value is `false`, then the run wouldn't actually have any system metrics enabled even though I called `mlflow.enable_system_metrics_logging()`.

Reply:
good point! I am changing `log_system_metrics` to default to `None`, then its value is of a higher priority over the environment variable. 

In another word, we will use arg `log_system_metrics` to determine if we should log system metrics, only if `log_system_metrics=None` we will check the environment variable. In this way, users would be able to have global control plus delicate control, e.g. "i want to globally turn off system metrics logging, but only for this run I want it to log system metrics". Does this sound reasonable?